### ISSUE TITLE
Request for a `softFindWithRef`

### ISSUE THREAD

**Issue by rjokelai** [sheepsteak/react-shallow-testutils#20]
A softer version of `findWithRef` would be beneficial when, e.g., hiding components:

```jsx
class Foo extends Component {
  render() {
    return (<div>
      {this.props.show ? <span ref="bar">Text</span> : undefined}
    </div>)
  }
}
```

With `<Foo show={false} />` the span will be hidden so `softFindWithRef(component, 'bar')` would return e.g., `undefined` instead of throwing error

**Comment by aaronbeall** [comment]
Agreed. However you could use `expect(findWithRef(elem, "bar")).toThrow()` to assert that the component wasn't renderered.

**Comment by aaronbeall** [comment]
Personally I would prefer to get back `null` than to throw, but you can use:

    expect(() => findWithRef(component, "bar").not.toThrow()

**Comment by sheepsteak** [comment]
The reason the functions that find a single result throw is to mirror how the [`react-addons-test-utils`](https://facebook.github.io/react/docs/test-utils.html#findrenderedcomponentwithtype) work. I use what @aaronbeall has put in his comment. In my opinion, either something should be there or it shouldn't and you want the test to fail as fast possible if you're wrong.
